### PR TITLE
Merge EtherCollateralsUSD and rest of Fomalhaut into develop

### PR DESCRIPTION
**Pull request by jacko125**

### CODE REVIEW

File: .solcover.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const {
 
 module.exports = {
 	port: 8545,
-	skipFiles: ['test-helpers', 'EscrowChecker.sol'],
+	skipFiles: ['test-helpers', 'EscrowChecker.sol', 'EtherCollateralsUSD.sol'],
```
Comment (ajsantander) [inline .solcover.js:7]: Why are we skipping these Solidity files from coverage?

File: contracts/FeePool.sol [hunk 2]
```diff
@@ -725,11 +733,12 @@ contract FeePool is Owned, Proxyable, SelfDestructible, LimitedSetup, MixinResol
...
 
     /* ========== Modifiers ========== */
-    modifier onlyExchangerOrSynth {
+    modifier onlyInternalContracts {
         bool isExchanger = msg.sender == address(exchanger());
         bool isSynth = issuer().synthsByAddress(msg.sender) != bytes32(0);
+        bool isEtherCollateralsUSD = msg.sender == address(etherCollateralsUSD());
 
-        require(isExchanger || isSynth, "Only Exchanger, Synths Authorised");
+        require(isExchanger || isSynth || isEtherCollateralsUSD, "Only Internal Contracts");
```
Comment (ajsantander) [inline contracts/FeePool.sol:741]: nit: Could save gas here by passing through to the function as soon as any condition is met.

**Review by ajsantander — CHANGES_REQUESTED** [review]
Looks good. Would like to know more about the situation with test coverage tho.

File: .solcover.js (hunk 1 above)
Comment (justinjmoses) [reply .solcover.js:7]: Because there is some issue in `EtherCollateralsUSD` coverage, it was decided to skip it.

**Review by ajsantander — DISMISSED** [review]

**Pull request merged**